test/fixtures/expression_ignored.html

```diff
@@ -1,4 +1,4 @@
 @{{ foo }}
-<p>
-  And here's a @{{ variable }} that would be undefined.
+<p data-username="@{{ user.name }}" data-user-id="user-@{{ user.id }}">
+  Here's one @{{ variable }} and here's @{{ another }}.
```

Comment:
non-blocking suggestion !

can we have one more example here with usual expression and one ignored expression on same element and also or the same attribute?

like this may 

```html
<p data-username="@{{ user.name }}" data-user-id="user-@{{ user.id }}-{{ user.id2 }}">
  And here's a @{{ variable }} that would be undefined.	  Here's one {{ variable }} and here's @{{ another }}.
```

ignore if already exists.

Reply:
Once `@{{` is encountered, the entire string/attribute value is returned as-is.

So this:

```
Here's one @{{ variable }} and here's some {{ foo }}.
```

will be output as:

```
Here's one {{ variable }} and here's some {{ foo }}.
```

So `{{ foo }}` will not be evaluated. Which come to think about it, may not be ideal, but I can't see how you'd evaluate `{{ foo }}`...?

Reply:
ummm, 

can we do like if `@{{` comes, it will ignore till first occurrence of `}}` ? WDYT ?

Reply:
I wonder, maybe changing the regexes here:

https://github.com/posthtml/posthtml-expressions/blob/77aaf5696593e7cd14ea41a6505e1f2d03ae8c6e/lib/index.js#L127-L138

... to match `{{ }}` or `{{{ }}}` as long as they don't start with `@`, would be better instead? So we only match non-ignored delimiters, and then we no longer have to return 'early', so my example above should (theoretically) correctly return:

```
Here's one {{ variable }} and here's some bar.
```

(assuming `foo: 'bar'` in `locals`).

Thoughts?

Reply:
> to match `{{ }}` or `{{{ }}}` as long as they don't start with `@`, would be better instead?

yeah, I think this should work.

Reply:
For example, something like:

```js
const delimitersRegexp = new RegExp(`[^@]${before}(.+?)${after}`, 'g')
```

And then, we'd replace any `@{{`` _after_ they were parsed by `placeholders`.

Reply:
`[^@]` why set ?

Reply:
So you don't match any `@{{ }}`. This way, we don't even need much of the code we currently have for handling `@{{ }}` expression ignoring.

Working on an example, just need to get the regex right, as it currently matches one character before `{{`, so it returns either ` {{` (with a space) or even `-{{` (with a dash in front) in the attribute example you suggested...

Reply:
Maybe you guys can help with the regex, I can't get it to _not_ match the character in front.

Given this:

```html
Here's one @{{ variable }} and here's some {{ foo }}.

<p data-username="@{{ user.name }}" data-user-id="user-@{{ user.id }}-{{ user.id2 }}"></p>
```

I need it to:

- match: `{{ foo }}`, `{{ user.id2 }}`
- not match: `@{{ variable }}`, `@{{ user.name }}`, `@{{ user.id }}`

I currently have `/[^@]{{(.+?)}}/g` but it matches ` {{ foo }}` (space char in front) and `-{{ user.id2 }}` (- in front).

Reply:
i meant we can use just `^@` without the set right ...but it would be a wrong regex as it would imply beginning char not the current seems fine 👍 

yeah, I think this your example regex will do the job

Reply:
It's not quite there yet, because you end up passing ` {{ foo }}` or `-{{ user.id2 }}` to `placeholders`, which will result in a `SyntaxError`.

Reply:
let me check if I can match the regex

Reply:
A negative lookbehind would work I think:

```
(?<!@){{(.+?)}}
```

And it's supported since Node 8.1.0, so I think it should work. Will test and confirm.

Reply:
Got it working, will push it up in a moment 👍

Reply:
can you try `^(?! )[^@]{(.+?)}}` ?

Reply:
great 👍

Reply:
So although I got it working, I think I found an issue with how `placeholders` treats unescaped delimiters. 

Basically, since it loops over every item in the `settings` object it receives, it'll still match `{{ var }}` inside `@{{{ var }}}`.

I'll push the fix for regular, escaped delimiters and will open a separate issue for unescaped delimiters.

Reply:
> I'll push the fix for regular, escaped delimiters and will open a separate issue for unescaped delimiters.

cool 👍